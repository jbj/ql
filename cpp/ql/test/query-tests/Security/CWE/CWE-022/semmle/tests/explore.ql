import semmle.code.cpp.ir.dataflow.DefaultTaintTracking as Default
import semmle.code.cpp.ir.IR
import semmle.code.cpp.ir.dataflow.DataFlow

query predicate tainted = Default::tainted/2;

from Default::DefaultTaintTrackingCfg cfg, Instruction i1, Instruction i2
where cfg.hasFlow(DataFlow::instructionNode(i1), DataFlow::instructionNode(i2))
select i1, i2