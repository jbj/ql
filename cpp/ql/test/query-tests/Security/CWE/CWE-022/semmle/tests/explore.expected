ERROR: Could not resolve type Default::DefaultTaintTrackingCfg (explore.ql:7,6-38)
ERROR: Query predicates cannot be annotated with cached (explore.ql:5,1-6)
